permissions/middlewares.js

```diff
@@ -0,0 +1,26 @@
+const { getActions } = require('./actions');
+const { asyncHandler } = require('../utils');
+const models = require('../models');
+
+const { Case, sequelize } = models;
+const CaseController = require('../controllers/case-controller')(Case, sequelize);
+
+const canEditCase = asyncHandler(async (req, res, next) => {
```

Comment:
A comment explaining the use of this function would be very useful. 
Could be something like this (feel free to expand it)
```
/**
 * Determines if the user can edit the case based on the actions triggered and the permissions.
 */
```